gtk2_ardour/sfdb_ui.cc

```diff
@@ -533,6 +533,10 @@ SoundFileBox::audition ()
 					SourceFactory::createExternal (DataType::AUDIO, *_session,
 											 path, n,
 											 Source::Flag (ARDOUR::AudioFileSource::NoPeakFile), false));
+				if (!afs) {
+					throw failed_constructor();
```

Comment:
not sure what you'd prefer, this or alternatively duplicating the code from the catch below?